client/state/utils.js

```diff
@@ -91,6 +93,14 @@ export const keyedReducer = ( keyName, reducer ) => {
 	const initialState = reducer( undefined, { type: '@@calypso/INIT' } );
 
 	return ( state = {}, action ) => {
+		if ( DESERIALIZE === action.type ) {
+			return deserializer( state );
+		}
+
+		if ( SERIALIZE === action.type ) {
+			return serializer( mapValues( state, item => reducer( item, action ) ) );
```

Comment:
this may be a silly worry, but this allocates a new object and enumerates the current object on every serialization pass. that's "slow" compared to the old default, just returning the current instance. For some things that have a large number of items in state (loaded reader posts, loaded sites, etc), that walk could get pretty spendy. 

so maybe handling this at a higher level is a better idea? If that higher level wanted to limit what's returned, it would still have to pay this cost. 

That said, I'm not sure how bad the cost really is. May not be an issue. Just caught my eye.